### DISCUSSION TITLE
Allow editing header in Daily widget

### DISCUSSION THREAD

**Discussion by danvolchek** [breezy-weather/breezy-weather#1226] [Ideas]
### Describe your idea

The daily widget (tile layout) is perfect for me: I love seeing the icon and then having configurable text next to it. That's so cool!!

I would also like to be able to edit the header/larger text. Specifically, I'd like to have the main temperature display be the feels like temperature, not the actual temperature.

Currently I put it in the subtitle, but it'd be nice if I could modify the header instead using the same placeholder system as the subtitle.

The text widget allows hiding the header and then only having a fully configurable subtitle, but it doesn't meet my needs because it doesn't show the icon or allow having a larger header/smaller subtitle.

This is similar to https://github.com/breezy-weather/breezy-weather/issues/446 but I would like to be able to edit the header text in any way, not only modify the temperature to be the feels like temperature - that is just what prompted this feature request.

This may be part of https://github.com/breezy-weather/breezy-weather/issues/937 already, but I wasn't sure after reading through it.

Also, thank you for making breezy weather :)

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.6 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
There are no plans for widget improvements besides what's included in #937.

We will design "fits most" widgets. If you want a widget tailored to your needs, your best bet is to design it and wait for #863 (or if it's good enough for use, just use already existing Gadgetbridge shared data).

**Reply by danvolchek** [reply]
Okay, thanks.

**Reply by danvolchek** [reply]
I don't understand how to use Gadgetbridge to make my own widget - it looks like it's an app for managing smart watches and things like that. Is there any documentation on how to do it?

**Reply by danvolchek** [reply]
Do you mean that Breezy Weather exports weather data in some format, and that data can be used by creating my own app with its own widgets?

That is more work than I'm willing to do, I'll just use the subtitle.

**Reply by papjul** [reply]
You don't need the Gadgetbridge app. We expose weather data through a receiver that was designed for this app, but any other app can use it.

It is exposed through the intent action `nodomain.freeyourgadget.gadgetbridge.ACTION_GENERIC_WEATHER` with the extra `WeatherJson` for the first location and/or `WeatherSecondaryJson` for the following locations. The content of the json can be decoded in the following class: https://github.com/breezy-weather/breezy-weather/blob/main/app/src/main/java/org/breezyweather/sources/gadgetbridge/json/GadgetbridgeData.kt

We also have plans to make our own design with a sample app so users can have something ready-to-use or understand how to use it, but there is no ETA.
